tests/test_draft_resource.py

```diff
@@ -79,4 +80,57 @@ def test_create_draft_of_existing_record(app, client, record_service,
     fields_to_check = ['pid', 'metadata', 'revision',
                        'created', 'updated', 'links']
 
+    for field in fields_to_check:
+        assert field in response_fields
+
     assert response.json['metadata']['title'] == orig_title
+
+
+def test_publish_draft_of_new_record(app, client, draft_service, input_record,
```

Comment:
We can get rid of draft_service it seems like!
```suggestion
def test_publish_draft_of_new_record(app, client, input_record,
```

I suspect we might be able to get rid of `app` too :fire: ...  `client` should push an application context by transitivity (https://github.com/pytest-dev/pytest-flask/blob/master/pytest_flask/fixtures.py#L16) making it unnecessary for us. (I saw the note below, so I know it has to be pushed). To be checked though.

Reply:
You were correct! 🔥 Nice catch :)